AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingSubscriptionEventPublisher.swift

```diff
@@ -9,7 +9,12 @@ import Amplify
 import AWSPluginsCore
 import Combine
 
+enum IncomingSubscriptionEventPublisherEvent {
+    case mutationEvent(MutationSync<AnyModel>)
+}
+
 @available(iOS 13.0, *)
 protocol IncomingSubscriptionEventPublisher {
-    var publisher: AnyPublisher<MutationSync<AnyModel>, DataStoreError> { get }
+    var publisher: AnyPublisher<IncomingSubscriptionEventPublisherEvent, DataStoreError> { get }
+    func kill()
```

Comment:
Let's move `kill()` into a `Killable` interface; or consider just renaming `kill()` to `Cancel` and have these publishers implement `Cancellable`. The contract would be that an invocation of `kill/cancel` is asynchronous, and would cause the receiver to emit a completion event on its publisher when finished disposing of its resources.